sdf/final-project/technical.md

```diff
@@ -6,102 +6,138 @@ This is intended to be filled out by staff members of DPI Tech Apprenticeship pr
+## Backend (max: 10 points)
+- [ ] **CRUD**: Does the application implement at least one resource with full CRUD functionality?
+- [ ] **MVC pattern**: Does the application follow the Model-View-Controller pattern, with skinny controllers and rich models?
+- [ ] **RESTful Routes**: Are the routes RESTful, with clear and consistent naming conventions?
+- [ ] **DRY queries**: Are database queries primarily implemented in the model layer rather than in the views, following the principle of keeping views lightweight and focused on presentation logic?
 - [ ] **Data Model Design**: Is the data model well-designed, clear, and efficient, facilitating easy data manipulation and retrieval, while avoiding redundancy and promoting data integrity?
-- [ ] **Security Practices**: Are security best practices followed, including secure storage of API keys and sensitive information (using .env or Rails credentials), enforcing HTTPS (config.force_ssl = true), and safeguarding against common vulnerabilities?
-- [ ] **Association and Query Optimization**: Does the application efficiently use Rails association methods (belongs_to, has_many, etc.) and scopes to organize data relationships and perform optimized database queries?
-- [ ] **Authorization and Access Control**: Is an authorization framework (such as Pundit) employed to manage user permissions and ensure secure access control across the application?
-
-### Bonus (max 2pts)
-- [ ] **Advanced Data Modeling**: Has the application utilized advanced ActiveRecord features, such as concerns, Active Record callbacks for model lifecycle management, or polymorphic associations to encapsulate business logic?
-- [ ] **Business Logic**: Does the application abstract logic into service objects when appropriate?
-- [ ] **Background Processing**: Are background jobs (eg ActiveJob) implemented for time-consuming processes, improving app performance and user experience?
-- [ ] **API Development**: Does the project provide a well-documented JSON API for integration with other services or frontend frameworks?
-- [ ] **Performance Optimization**: Is the Bullet gem (or similar) used in development to detect and reduce N+1 queries and other common performance bottlenecks?
+- [ ] **Associations**: Does the application efficiently use Rails association methods (belongs_to, has_many, etc.) to organize data relationships?
+- [ ] **Validations**: Are validations implemented to ensure data integrity and consistency?
+- [ ] **Query Optimization**: Does the application use scopes to perform optimized database queries?
 - [ ] **Database Management**: Are additional features such as file upload (CSV) or custom rake tasks for database management included?
+- [ ] **Testing**: Does the project include a comprehensive test suite (e.g., RSpec, Minitest) that covers models, controllers, and other critical components?
```

Comment:
i don't think we need "comprehensive". Maybe tests for key flows or logic?

Reply:
fixed